File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/video/QualityImpl.kt
package dev.hebei.camerax_android.video

import dev.hebei.camerax_android.CameraXRegistrar
import dev.hebei.camerax_android.PigeonApiQualityApi

class QualityImpl(registrar: CameraXRegistrar) : PigeonApiQualityApi(registrar) {
    override fun fhd(): androidx.camera.video.Quality {
        return androidx.camera.video.Quality.FHD
    }

    override fun hd(): androidx.camera.video.Quality {
        return androidx.camera.video.Quality.HD
    }

    override fun highest(): androidx.camera.video.Quality {
        return androidx.camera.video.Quality.HIGHEST
    }

    override fun lowest(): androidx.camera.video.Quality {
        return androidx.camera.video.Quality.LOWEST
    }

    override fun sd(): androidx.camera.video.Quality {
        return androidx.camera.video.Quality.SD
    }

    override fun uhd(): androidx.camera.video.Quality {
        return androidx.camera.video.Quality.UHD
    }
}

File: camerax_android/lib/src/video/quality_impl.dart
import 'package:camerax_android/src/camerax_api.g.dart';
import 'package:camerax_platform_interface/camerax_platform_interface.dart';

final class QualityImpl extends Quality {
  static QualityImpl get sd => QualityImpl.internal(QualityProxyApi.sd);
  static QualityImpl get hd => QualityImpl.internal(QualityProxyApi.hd);
  static QualityImpl get fhd => QualityImpl.internal(QualityProxyApi.fhd);
  static QualityImpl get uhd => QualityImpl.internal(QualityProxyApi.uhd);
  static QualityImpl get lowest => QualityImpl.internal(QualityProxyApi.lowest);
  static QualityImpl get highest => QualityImpl.internal(QualityProxyApi.highest);

  final QualityProxyApi api;

  QualityImpl.internal(this.api) : super.impl();
}

extension QualityX on Quality {
  QualityProxyApi get api {
    final impl = this;
    if (impl is! QualityImpl) throw TypeError();
    return impl.api;
  }
}

extension QualityProxyApiX on QualityProxyApi {
  Quality get impl => QualityImpl.internal(this);
}
